src/Storages/MergeTree/registerStorageMergeTree.cpp

```diff
@@ -620,6 +615,28 @@ static StoragePtr create(const StorageFactory::Arguments & args)
                 args.getLocalContext()->checkMergeTreeSettingsConstraints(initial_storage_settings, storage_settings->changes());
             metadata.settings_changes = args.storage_def->settings->ptr();
         }
+
+        auto constraints = metadata.constraints.getConstraints();
```

Comment:
I put this below because `add_implicit_sign_column_constraint_for_collapsing_engine` can be enabled from the `SETTINGS` arg in a query, and the parsing of that only happens on line 612 just above this block. 

So if we put this block before line 612, then we need to check both `storage_settings` and `args.storage_def->settings`.